app-ios/tutanota/Sources/Alarms/AlarmPersistor.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+protocol AlarmPersistor {
+  var alarms: [EncryptedAlarmNotification] { get }
+  
+  func clear()
+  
+  func store(alarms: [EncryptedAlarmNotification])
+}
+
+class AlarmPreferencePersistor : AlarmPersistor {
+  var alarms: [EncryptedAlarmNotification] {
+    get {
+      return userPreferenceFacade.alarms
+    }
+  }
+  
+  let userPreferenceFacade: UserPreferenceFacade
+  let keychainManager: KeychainManager
+  
+  init (userPreferenceFacade: UserPreferenceFacade, keychainManager: KeychainManager) {
+    self.userPreferenceFacade = userPreferenceFacade
+    self.keychainManager = keychainManager
+  }
+  
+  func clear() {
+    self.userPreferenceFacade.clear()
```

Comment:
are we not storing anything else in the userprefs?

Reply:
just saw that the userPreferenceFacade is only responsible for the alarm and sse stuff, nvm. maybe we should rename it or use it in all places where we access userprefs

Reply:
You are right, it only stores alarm and SSE, it is SSE storage and alarm storage rolled up into one inefficient blob. Do you have a good name?

Reply:
Maybe `NotifcationStorage` as we don't reeeeeally do SSE and it's more than alarms? Although it's only needed for alarms